### ISSUE TITLE
Combinators and Ampersand and UnnecessaryParentReference

### ISSUE THREAD

**Issue by ifyoumakeit** [brigade/scss-lint#594]
SCSS shouldn't really compile `~ &` because it's doesn't do anything but `UnnecessaryParentReference` leads developers to drop the ampersands proceeding a combinator, despite it not doing what it's supposed to do in the `~ &` case.  The parent selector is nice, but maybe there should be an additional rule to use `& ~ &` when the second value starts with an ampersand.

**SCSS**
```scss
.foo {  
  & ~ & {
    background: green; // Valid
  }
  ~ & {
    background: black; // Invalid
  }
}
```
**CSS**
```css
.foo ~ .foo {
  background: green;
}
~ .foo {
  background: black;
}
```

Maybe `NecessaryParentReference`?

**Comment by ifyoumakeit** [comment]
Edited and changed title for clarity.

**Comment by sds** [comment]
Hey @ifyoumakeit, this request makes sense to me. It's an quirky behavior of Sass that ultimately makes sense but is initially confusing. Thanks for the suggestion&mdash;happy to merge a pull request with the implementation.

**Comment by wouter-muller** [comment]
+1

I have a similar case where i do the following:
```
  li {
    .....

    & + li::before {
      .....
    }
  }
```
SCSS-lint is complaining about the use of `li` as it thinks it is an unnecessary parent selector, but if i replace it with `&` my CSS doesn't work anymore.

**Comment by Mifrill** [comment]
btw, thanks for this linter, it's really helpful!

I've also have similar issue with `ember-component-css`

the rule of usage is:
```
& {  // ampersand refers to the component itself (parent selector)
  padding: 2px;
}
```
https://github.com/ebryn/ember-component-css#usage

but linter is raised the offense:
```
Unnecessary parent selector (&) 
```
well it's broke the build because CC is using this linter:
https://docs.codeclimate.com/docs/scss-lint
